### PR TITLE
Test

### LINKED ISSUE #1: create a parser compatible with eslint-typescript-parser 6.0
Remove all @xxx annotations before pass the source code string to eslint-typescript-parser

### PR DESCRIPTION
**Pull request by KKiiim**

### CODE REVIEW

File: asParser.cjs [hunk 1]
```diff
@@ -1,35 +1,66 @@
...
-                newCode += ' ';
-                break;
-            }else{
-                newCode += text[i];
-            }
-        }else{
-            newCode += text[i];
-        }
+  var arr = new Array();
+  for (let i = 0; i < text.length; i++) {
```
Comment (Changqing-JING) [inline asParser.cjs:18]: refactor to a class using getNextChar

**Review by Changqing-JING — CHANGES_REQUESTED** [review]

File: src/asParser.cjs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+function parse(code, options) {
+  let myParser = new asParser(code);
+  return tsParse.parse(myParser.getresult(), options);
+}
+
+class asParser{
+  constructor(code){
+    this.sourceCode = code;
+    this.tmpArr = new Array();
+    this.newCode = "";
```
Comment (Changqing-JING) [inline src/asParser.cjs:12]: don't need to be class member

File: src/asParser.cjs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    let newChar = this.sourceCode[this.index];
+    this.index++;
+    return newChar;
+  }
+
+  atProcessor(){
+    this.tmpArr.push(" ");
+    while (this.index < this.sourceCode.length) {
+      let tmpC = this.getNextChar();
+      if(tmpC == "\n" || tmpC == " "){
```
Comment (Changqing-JING) [inline src/asParser.cjs:39]: consider also tab

File: package.json [hunk 4]
```diff
@@ -4,19 +4,17 @@
...
+  "main": "./src/asParser.cjs",
+  "exports": "./src/asParser.cjs",
   "scripts": {
-    "lint": "npm-run-all \"lint:*\"",
     "lint:eslint-docs": "npm-run-all \"update:eslint-docs -- --check\"",
     "lint:js": "eslint .",
     "test": "mocha tests --recursive",
     "update:eslint-docs": "eslint-doc-generator",
-    "eslint-prettier": "npx prettier . --write && eslint --ext .js,.vue --fix && npx prettier . --check"
+    "lint": "npx prettier . --write && eslint --ext .js,.vue --fix && npx prettier . --check"
```
Comment (Changqing-JING) [inline package.json:17]: no npx prettier . --write
no vue
no npx prettier . --check

**Review by Changqing-JING — CHANGES_REQUESTED** [review]

File: tsconfig.json [hunk 5]
```diff
@@ -1,16 +1,15 @@
 {
   "compilerOptions": {
+    "target": "ES2021" /* Set the JavaScript language version for emitted JavaScript and include compatible library declarations. */,
```
Comment (Changqing-JING) [inline tsconfig.json:3]: remove this file

**Review by Changqing-JING — CHANGES_REQUESTED** [review]

File: package.json [hunk 6]
```diff
@@ -27,7 +26,17 @@
     "eslint-plugin-eslint-plugin": "^5.0.0",
     "eslint-plugin-node": "^11.1.0",
     "mocha": "^10.0.0",
-    "npm-run-all": "^4.1.5"
+    "npm-run-all": "^4.1.5",
+    "@types/chai": "^4.3.9",
```
Comment (Changqing-JING) [inline package.json:30]: cleanup unnecessary deps

File: testfile/exports-lazy.ts [hunk 7]
```diff
@@ -0,0 +1,9 @@
+@lazy export const lazyGlobalUnused: i32[] = [1,2,3];
```
Comment (Changqing-JING) [inline testfile/exports-lazy.ts:1]: move testfile to tests folder

**Review by Changqing-JING — CHANGES_REQUESTED** [review]

File: .npmignore [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+node_modules
+tests
+testfile
+.eslintrc.cjs
+.gitignore
+.prettierignore
+.prettierrc.yaml
+.prettierrc
+package-lock.json
+README.md
```
Comment (Changqing-JING) [inline .npmignore:11]: remove readme

**Review by Changqing-JING — CHANGES_REQUESTED** [review]

File: package.json [hunk 9]
```diff
@@ -9,15 +9,16 @@
   "author": "Kim",
   "main": "./src/asParser.cjs",
   "scripts": {
-    "test": "node tests/asfileTest.cjs",
+    "test": "mocha tests/asfileTest.cjs",
```
Comment (Changqing-JING) [inline package.json:12]: run all tests under tests folder with mocha

**Review by Changqing-JING — CHANGES_REQUESTED** [review]

**Review by Changqing-JING — APPROVED** [review]

**Review by XMadrid — APPROVED** [review]

**Pull request merged**